### ISSUE TITLE
Bug: Scope manager: Catch clause scopes' `def`s have pattern as `name` when param is `ObjectPattern` or `ArrayPattern`

### ISSUE THREAD

**Issue by overlookmotel** [typescript-eslint/typescript-eslint#11981]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

scope-manager

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&sourceType=script&showAST=scope&fileType=.js&code=C4JwngBA3gvhDGBDY8AWEAUUKIDQQA8AuCAI3zBIG159sATCGAXXwFMIBeCARiYEpoMAFDDQkWAmRpMVAGb4qAc3yo6EAJb4AXiQBWXCACYmzZoNjCgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
try {} catch ({ a, x: b, y: [c, { d }], e = 1 }) {}

try {} catch ([f, [g, h, { i, z: j = 2 }]]) {}
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

Expected the `name` property of each of the `defs` for bindings in the catch clause param to be the `Identifier`s for those bindings, matching ESLint's behavior.

That behavior is also consistent with TS-ESLint for other places where bindings are defined via destructuring e.g.:

```js
const { a } = foo();
const [ b ] = bar();
function f({ c }) {}
```

### Actual Result

`def.name` is the catch clause param node (`ObjectPattern` or `ArrayPattern`).

### Additional Info

A small ESLint plugin which demonstrates that ESLint has `def.name` as the `Identifier` for each binding:

```js
export default {
  rules: {
    'check-catch-scopes': {
      create(context) {
        for (const scope of context.sourceCode.scopeManager.scopes) {
          checkScope(scope);
        }

        return {};
      },
    },
  },
};

function checkScope(scope) {
  if (scope.type !== "catch") return;

  for (const variable of scope.variables) {
    const { name } = variable;
    const error = checkVar(variable);
    if (error) {
      console.log("FAIL:", name, error);
    } else {
      console.log("Pass:", name);
    }
  }
}

function checkVar(variable) {
  const { defs, identifiers } = variable;
  if (defs.length !== 1) return "More than 1 def";
  if (identifiers.length !== 1) return "More than 1 identifier";

  const defName = defs[0].name;
  if (defName.type !== "Identifier") return "Not Identifier";
  const ident = identifiers[0];
  if (defName.start !== ident.start || defName.end !== ident.end) return "Span mismatch";
}
```

Run on the above source (without TS-ESLint), it logs:

```
Pass: a
Pass: b
Pass: c
Pass: d
Pass: e
Pass: f
Pass: g
Pass: h
Pass: i
Pass: j
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/scope-manager`     | `8.53.1` |

**Comment by bradzacher** [comment]
why in the name of heck is this valid syntax?
who would ever want to destructure the catch variable?
what were they thinking making that allowed?!?

that is _literally_ insane.

**Comment by overlookmotel** [comment]
I'm trying to pass 100% of ESLint's tests, and this is one of them!

Actually, it's not *completely* bizarre. Have seen this in the wild:

```js
try {
  // ...
} catch ({ message }) {
  // ...
}
```

**Comment by overlookmotel** [comment]
PR: #11982
